src/pathfinding_service/typing.py

```diff
@@ -15,4 +16,10 @@ class AddressReachabilityProtocol(Protocol):
     def get_address_reachability(self, address: Address) -> AddressReachability:
         ...
 
+    def get_userid_presence(self, user_id: str) -> UserPresence:
+        ...
+
+    def get_userids_for_address(self, address: Address) -> Set[str]:
+        ...
+
     _address_to_userids: Dict[Address, Set[str]]
```

Comment:
Now that you have added the functions above, `_address_to_userids` can be removed from this protocol.